jenkins/spark-nightly-build.sh

```diff
@@ -21,36 +21,80 @@ set -ex
-mvn -U -B -Dbuildver=321 clean install -pl '!tools' $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER
-[[ $SKIP_DEPLOY != 'true' ]] && mvn -B -Dbuildver=321 deploy -pl '!tools,!dist' $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DskipTests
-# temporarily skip tests on Spark 3.3.0 - https://github.com/NVIDIA/spark-rapids/issues/4031
-mvn -U -B -Dbuildver=330 clean install -pl '!tools' $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DskipTests
-[[ $SKIP_DEPLOY != 'true' ]] && mvn -B -Dbuildver=330 deploy -pl '!tools,!dist' $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER -DskipTests
+
+for buildver in "${SPARK_SHIM_VERSIONS[@]:1}"; do
+    # temporarily skip tests on Spark 3.3.0 - https://github.com/NVIDIA/spark-rapids/issues/4031
+    [[ buildver == "330" ]] && skipTestsFor330=true || skipTestsFor330=false
+    mvn -U -B clean install -pl '!tools' $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR \
+        -Dcuda.version=$CUDA_CLASSIFIER \
+        -Dbuildver="${buildver}" \
+        -DskipTests="${skipTestsFor330}"
+    # fix up dist jar with reduced pom in the local m2
+    mvn -B install:install-file $MVN_URM_MIRROR \
+        -Dcuda.version=$CUDA_CLASSIFIER \
+        -Dmaven.repo.local=$M2DIR \
+        -Dfile="${DIST_FPATH}.jar" \
+        -DgroupId="${ART_GROUP_ID}" \
+        -DartifactId="${ART_ID}" \
+        -Dversion="${ART_VER}" \
+        -Dpackaging=jar
+    [[ $SKIP_DEPLOY != 'true' ]] && \
+        mvn -B deploy -pl '!tools,!dist' $MVN_URM_MIRROR \
+            -Dmaven.repo.local=$M2DIR \
+            -Dcuda.version=$CUDA_CLASSIFIER \
+            -DskipTests \
+            -Dbuildver="${buildVer}"
+done
+
+mvn -B clean install -pl '!tools' \
+    -PsnapshotsWithDatabricks \
+    -Dbuildver=$SPARK_BASE_SHIM_VERSION \
+    $MVN_URM_MIRROR \
+    -Dmaven.repo.local=$M2DIR \
+    -Dcuda.version=$CUDA_CLASSIFIER
+
+# fix up dist jar with reduced pom in the local m2
+mvn -B install:install-file $MVN_URM_MIRROR \
+    -Dcuda.version=$CUDA_CLASSIFIER \
+    -Dmaven.repo.local=$M2DIR \
+    -Dfile="${DIST_FPATH}.jar" \
+    -DgroupId="${ART_GROUP_ID}" \
+    -DartifactId="${ART_ID}" \
+    -Dversion="${ART_VER}" \
+    -Dpackaging=jar
 
 if [[ $SKIP_DEPLOY != 'true' ]]; then
-    mvn -B -pl '!tools' -Dbuildver=301 -PsnapshotsWithDatabricks clean deploy $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER
-else
-    mvn -B -pl '!tools' -Dbuildver=301 -PsnapshotsWithDatabricks clean install $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR -Dcuda.version=$CUDA_CLASSIFIER
+    # deploy dist with reduced pom
+    mvn -B deploy:deploy-file
+        $MVN_URM_MIRROR -Dmaven.repo.local=$M2DIR \
+        -Dcuda.version=$CUDA_CLASSIFIER \
+        -Dfile="${DIST_FPATH}.jar" \
+        -DgroupId="${ART_GROUP_ID}" \
+        -DartifactId="${ART_ID}" \
```

Comment:
aren't we missing the actual reduced pom here?  -DpomFile="$POM_FPATH"
            
Maybe I'm missing something else?

Reply:
we rely on Maven features quoted in the description to use the pom inside the jar that is now correct

Reply:
for future reference, I ended up putting -DpomFile back because upgrading the plugins to 3.0.0-M1 is triggering cascading changes to unrelated places in the build